src/gui/sdlmain.cpp

```diff
@@ -1063,27 +1085,25 @@ static void maybe_present_synced(const bool present_if_last_skipped)
 	last_sync_time = should_present ? GetTicksUs() : now;
 }
 
-// Present new frames, but if not new, re-present the last frame up to 10 times
-// (as back-to-back duplicates) to ensure the host shows it as opposed to
-// showing a prior frame in the event the host dropped the original when we
-// first presented it.
 static void present_new_or_maybe_dupe(const bool frame_is_new)
 {
-	constexpr auto max_dupes = 10;
-	static auto remaining_dupes = 0;
+	// The tempo limits dulicate presentation to every N calls of this
+	// function (the backfill_tempo). It's meant to produce low single-digit
+	// Hz regardless of the DOS rate.
+	static auto backfill_countdown = sdl.frame.vfr_backfill_tempo;
 
 	if (frame_is_new) {
 		sdl.frame.present();
-		remaining_dupes = max_dupes;
-	}
-	// We're now presenting duplicate frames - but be less agressive with
-	// them by using the sync'd presenter to run the dupes at the host's
-	// rate to avoid causing further blocking / stuttering / dropped frames
-	// on the host-side.
-	else if (remaining_dupes > 0) {
+		backfill_countdown = sdl.frame.vfr_backfill_tempo;
+	} else if (--backfill_countdown <= 0) {
+		// The duplicate presenter is an insurance policy to ensure the
+		// last frame is shown even if the host previously dropped it.
+		// This is a temporary hack; it should be removed in the future
+		// when everyone's PC display technology can support > 70 Hz
+		// circa-1980s CRT rates.
```

Comment:
Probably this comment will stay here until 2030 then 😅

Reply:
Extra fussy remark: 80s monitors were more like small TVs, so 50Hz in PAL regions and 60Hz in NTSC land (for home computers, and for Hercules to EGA). 

From my point of view, it was VGA that brought 70Hz screen modes "to the masses", which happened in early 90s. But sure, it was invented in 1987 😄 

So I'd say "1990s PC CRT monitors".

Sorry, I'm really into this thing 😅

Reply:
Nice clarification :-) Added.